pkg/apis/batch/validation/validation.go

```diff
@@ -237,6 +243,18 @@ func validatePodFailurePolicy(spec *batch.JobSpec, fldPath *field.Path) field.Er
 	return allErrs
 }
 
+func validatePodReplacementPolicy(spec *batch.JobSpec, onReplacePath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if spec.PodFailurePolicy != nil && spec.PodReplacementPolicy != nil {
```

Comment:
nit: I think it will read easier if we had an outer check for `spec.PodReplacementPolicy != nil`.